Cython/Shadow.pyi

```diff
@@ -1,44 +1,247 @@
+import dataclasses
 from builtins import (int as py_int, float as py_float,
                       bool as py_bool, str as py_str, complex as py_complex)
-from typing import (Union, Dict, Any, Sequence, Optional,
-                    List, TypeVar, Type, Generic)
+from types import TracebackType
+from typing import (Any, Iterable, Literal, ParamSpec, Sequence, Optional, Type,
+                    TypeVar, Generic, Callable, final, overload)
+
+# This is necessary so that type checkers don't ignore the 'dataclasses' import
+# or the 'final' import from typing.
+__all__ = (
+    '__version__',
+    'dataclasses', 'final',  # from imports above
+    'int', 'long', 'longlong', 'short', 'char', 'sint', 'slong', 'slonglong',
+    'sshort', 'schar', 'uint', 'ulong', 'ulonglong', 'ushort', 'uchar',
+    'size_t', 'Py_ssize_t',
+    'Py_UCS4', 'Py_UNICODE',
+    'float', 'double', 'longdouble',
+    'complex', 'floatcomplex', 'doublecomplex', 'longdoublecomplex',
+    'bint', 'void', 'basestring', 'unicode',
+    'gs', 'compiled',
+    'cfunc', 'ccall', 'compile', 'locals', 'returns', 'exceptval',
+    'cclass', 'internal', 'c_api_binop_methods', 'type_version_tag', 'no_gc_clear', 'no_gc',
+    'boundscheck', 'wraparound', 'initializedcheck', 'nonecheck', 'cdivision', 'cdivision_warnings', 'profile',
+    'linetrace', 'infer_types', 'emit_code_comments',
+    'binding', 'embedsignature', 'always_allow_keywords', 'unraisable_tracebacks', 'iterable_coroutine', 'cpp_locals',
+    'overflowcheck', 'optimize', 'warn',
+    'inline', 'cdiv', 'cmod', 'cast', 'sizeof', 'typeof', 'address', 'declare',
+    'nogil', 'gil',
+    'CythonTypeObject', 'CythonType', 'PointerType', 'ArrayType',
+    'pointer', 'array', 'struct', 'union', 'typedef', 'fused_type'
+)
```

Comment:
`__all__` is usually spelled as a list, not a tuple, because it's a sequence of similar objects, not a struct-like value.
But it's unfortunate that this is needed at all, because it has a high chance of being forgotten in future changes.
It it's really mostly about the `dataclasses` import, can't we just declare that as `dataclasses: object` here?
```suggestion
__all__ = [
    '__version__',
    'dataclasses', 'final',  # from imports above
    'int', 'long', 'longlong', 'short', 'char', 'sint', 'slong', 'slonglong',
    'sshort', 'schar', 'uint', 'ulong', 'ulonglong', 'ushort', 'uchar',
    'size_t', 'Py_ssize_t',
    'Py_UCS4', 'Py_UNICODE',
    'float', 'double', 'longdouble',
    'complex', 'floatcomplex', 'doublecomplex', 'longdoublecomplex',
    'bint', 'void', 'basestring', 'unicode',
    'gs', 'compiled',
    'cfunc', 'ccall', 'compile', 'locals', 'returns', 'exceptval',
    'cclass', 'internal', 'c_api_binop_methods', 'type_version_tag', 'no_gc_clear', 'no_gc',
    'boundscheck', 'wraparound', 'initializedcheck', 'nonecheck', 'cdivision', 'cdivision_warnings', 'profile',
    'linetrace', 'infer_types', 'emit_code_comments',
    'binding', 'embedsignature', 'always_allow_keywords', 'unraisable_tracebacks', 'iterable_coroutine', 'cpp_locals',
    'overflowcheck', 'optimize', 'warn',
    'inline', 'cdiv', 'cmod', 'cast', 'sizeof', 'typeof', 'address', 'declare',
    'nogil', 'gil',
    'CythonTypeObject', 'CythonType', 'PointerType', 'ArrayType',
    'pointer', 'array', 'struct', 'union', 'typedef', 'fused_type'
]
```

Reply:
Yes I agreee, this was a very unfortunate decision I had to make. I took another stab at this today and found this note in [PEP 484](https://peps.python.org/pep-0484/#stub-files):
> Modules and variables imported into the stub are not considered exported from the stub unless the import uses the `import ... as ...` form or the equivalent from `... import ... as ...` form. (*UPDATE*: To clarify, the intention here is that only names imported using the form X as X will be exported, i.e. the name before and after as must be the same.)

See 707222c58208919f1c93510fb23f598a418bc0a9